docs/using_eessi/building_on_eessi.md

```diff
@@ -6,67 +6,113 @@ Building on top of EESSI with EasyBuild is relatively straightforward. One cruci
 ### Starting the EESSI software environment
 Start your environment as described [here](../using_eessi/setting_up_environment.md)
 
-### Configure EasyBuild
-To configure EasyBuild, first, check out the [EESSI software-layer repository](https://github.com/EESSI/software-layer.git). We advise you to check out the branch corresponding to the version of EESSI you would like to use.
+### Using the EESSI-extend module
+The `EESSI-extend` module facilitates building on top of EESSI using EasyBuild. It does a few key things:
 
-If you are unsure which version you are using, you can run
+1. It configures EasyBuild to match how the rest of the EESSI software is build
+2. It configures EasyBuild to use a certain installation path (e.g. in your homedir), taking into account the hardware architecture you are building on
+3. It adds the relevant subdirectory from your installation path to your `MODULEPATH`, to make sure your newly installed modules are available
+4. It loads the EasyBuild module
+
+The `EESSI-extend` module recognizes a few environment variables. To print an up-to-date list, check the module itself
 ```
-echo ${EESSI_VERSION}
+module help EESSI-extend/2023.06-easybuild
 ```
-to check it.
 
-To build on top of e.g. version `2023.06` of the EESSI software stack, we check it out, and go into that directory:
+The key environment variables it will respect are:
+ - EESSI_USER_INSTALL can be set to a location to install modules for use by the user only. The location must already exist on the filesystem. Set this environment variable if you're an end-user who wants to install additional software just for yourself.
+ - EESSI_PROJECT_INSTALL can be set to a location to install modules for use by a project. The location must already exist on the filesystem and you should ensure that the location has the correct Linux group and the SGID permission is set on that directory (`chmod g+s $EESSI_PROJECT_INSTALL`) so that all members of the group have permission to read and write installations. Set this environment variable if you're an end-user who's part of a project, and wants to create additional installations that should be available to all project members.
+ - EESSI_SITE_INSTALL is either defined or not and cannot be used with another environment variable. A site installation is done in a defined location and any installations there are (by default) world readable. Set this environmnet variable if you are a site hosting a system that has EESSI available, and you want to build on top to make additional modules available to all of your users.
+ - EESSI_CVMFS_INSTALL is either defined or not and cannot be used with another environment variable. A CVMFS installation targets a defined location which will be ingested into CVMFS and is only useful for CVMFS administrators. Set this environment variable if you manage your own CVMFS repository where the software has to be built on top of EESSI.
```

Comment:
This is true, but it is not really of relevance here...

Reply:
Ehm, what's the 'this' here exactly? I'm assuming it's relevant to explain the environment variables that can be set here, no?

Reply:
`EESSI_CVMFS_INSTALL`, it can only really work for us (who can configure write access to that location)

Reply:
What if I host my own Stratum 0 at `software.surf.nl`, and use a (the) build container to mount that with a writeable overlay. Can I then build _on top of EESSI_ in my own CVMFS repo?

Reply:
Yes, you can do that today by using `EESSI_PROJECT_INSTALL` but that won't give you world readable rights on the installations.

Other scenarios could be supported, we'd just need the specific use case and then implement the logic. For what you mention, you really just want `EESSI_PROJECT_INSTALL` but with world readable permissions.

Reply:
Ok, well, something for later, we're note that far yet anyway :)